tldw_Server_API/app/core/DB_Management/media_db/repositories/media_repository.py

```
@@ -0,0 +1,864 @@
+
+class MediaRepository:
+    """Caller-facing seam for media persistence while internals migrate out of the shim."""
+
+    def __init__(self, session: MediaDbLike):
+        self.session = session
+
+    @classmethod
+    def from_legacy_db(cls, db: MediaDbLike) -> "MediaRepository":
+        return cls(session=db)
+
+    def add_media_with_keywords(
```

Ruff (0.15.6)

[warning] Too many return statements (7 > 6)

(PLR0911)